guides/plugins/plugins/content/cms/real-world-cms/add-real-world-cms-element.md

```diff
@@ -0,0 +1,460 @@
+```javascript
+import template from './sw-cms-el-preview-cms-button.html.twig';
+import './sw-cms-el-preview-cms-button.scss';
+
+const { Component } = Shopware;
+
+Component.register('sw-cms-el-preview-cms-button', {
+    template
+});
+```
+{% endcode %}
+
+We registered the preview now we only have to do the template and some styles. So we create a new file `sw-cms-el-preview-cms-button.html.twig`.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/preview/sw-cms-el-preview-cms-button.html.twig" %}
+{% raw %}
+```markup
+{% block sw_cms_element_ninja_button_preview %}
+	<div class="ninja-button-preview">
+		<button class="ninja-btn">Button</button>
+	</div>
+{% endblock %}
+
+```
+{% endraw %}
+{% endcode %}
+
+We only display a simple button right here. Let's add some styles so people can actually take a look at it:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/preview/sw-cms-el-preview-cms-button.scss" %}
+```css
+.ninja-button-preview {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 100%;
+
+    .ninja-btn {
+        background-color: #4492ed;
+        border: none;
+        color: #fff;
+        padding: 14px 50px;
+        text-align: center;
+        text-decoration: none;
+        display: inline-block;
+        font-size: 16px;
+        margin: 30px 0;
+    }
+}
+```
+{% endcode %}
+
+
+### Element config component
+
+The last thing you need for the CMS-Element in the administration is the config. And it's about the same here as well. We create a `config` directory and place 3 different files in here. Let's start with the `index.js`
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/config/index.js" %}
+```javascript
+import template from './sw-cms-el-config-cms-button.html.twig';
+
+const { Component, Mixin } = Shopware;
+
+
+Component.register('sw-cms-el-config-cms-button', {
+    template,
+
+    inject: ['repositoryFactory'],
+
+    mixins: [
+        Mixin.getByName('cms-element')
+    ],
+
+    created() {
+        this.createdComponent();
+    },
+
+    methods: {
+        createdComponent() {
+            this.initElementConfig('cms-button');
+        },
+
+        onElementUpdate(element) {
+            this.$emit('element-update', element);
+        },
+        onInputText(title) {
+            this.emitChanges(title);
+        }
+    }
+
+});
+```
+{% endcode %}
+
+Let's create the twig file as well real quick so we have the option to actually configure the button:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/elements/cms-button/config/sw-cms-el-config-cms-button.html.twig" %}
+{% raw %}
+```markup
+{% block sw_cms_el_ninja_cms_button %}
+	<sw-tabs class="sw-cms-el-config-ninja-cms-button__tabs" defaultItem="content">
```

Comment:
Mixed indention

Reply:
Indentation... :D sorry

Reply:
That's my bad I did disable my linter/formatting tool. Running a tool above the files probably should fix the issue I guess?